_posts/2024-08-14-adding-to-bioconda-quickguide.md

```diff
@@ -0,0 +1,468 @@
+title: Adding bioinformatic software to Bioconda - a reference guide
+author: "James A. Fellows Yates"
+tags:
+  - development
+  - conda
+  - bioconda
+  - software
+contributors:
+  - jfy133
+---
+
+This post aims to provide an (opinionated) guide to adding a tool to Bioconda, and how to debug Bioconda and the associated Biocontainer builds.
+
+The [conda package manager](https://docs.conda.io/en/latest/) combined the [Bioconda](https://bioconda.github.io/) repository has become a _de facto_ gold-standard way for distribute bioinformatics software ({% cite Gruening2018 %}).
+The associated [Biocontainer](https://biocontainers.pro/) project serves to provide complementary Docker and Singularity containers from the same conda ({% cite Da_Veiga_Leprevost2017 %}).
+
+While the Bioconda team has made a huge amount of impressive infrastructure to make adding our bioinformatic tools and packages to the repository as easy as possible, the documentation is lacking at points.
+In particular, while the [Bioconda documentation](https://bioconda.github.io/contributor/index.html) explains nicely how to contribute to the repository, I personally found that it misses out on the important part of how to make what we _actually_ will add i.e, how to _make_ the [conda recipe](https://docs.conda.io/projects/conda-build/en/latest/concepts/recipe.html) specifically for Bioconda [^1]. Another thing I found particularly tricky is how to debug the build process if things go wrong.
+
+I hope to here to provide guidance, based on my personal experience, to people who are interested to adding their software to Bioconda but maybe felt it too overwhelming to start.
+Hopefully it will provide sufficient 'hand-holding' to get us through the process, after which it's just 'rinse and repeat'!
+
+The main sections of this post are:
+
+- TL;DR
+- Prerequisites
+- Adding a new tool or package
+- Debugging a recipe build
+- Updating an existing tool or package recipe
+
+Please note that this post comes with 'no warranty'!(!), as the Bioconda build steps could change at any point.
+However, the steps here should act as a good starting point.
+Furthermore, if you are planning to add someone else's tool or package to Bioconda, it's always good etiquette to ask or just inform the original authors that it will happen.
+
+_I would like to thank George Bouras (@gbouras13) for prompting the formalisation of my rough notes that he shared on Twitter where it became more popular than I expected._
+
+## TL;DR
+
+### Overview
+
+![Overview diagram of the steps of the steps described in this post.]({% link assets/images/2024-08-14-bioconda-guide/bioconda-guide-workflowdiagram_short.svg %})
+
+### Relevant Commands
+
+```bash
+## Create environment with conda building tools
+conda create -n bioconda-build -c conda-forge -c bioconda conda-build bioconda-utils greyskull
+conda activate bioconda-build
+
+## Clone repo and make branch
+git clone <your-forked-bioconda-recipes-repo-address>
```

Comment:
include link to bioconda-recipes-repo in TLDR https://github.com/bioconda/bioconda-recipes